client/my-sites/checkout/checkout/incompatible-product-message.tsx

```diff
@@ -0,0 +1,28 @@
+/**
+ * External dependencies
+ */
+import React, { FC } from 'react';
+import type { TranslateResult } from 'i18n-calypso';
+
+/**
+ * Internal dependencies
+ */
+import Gridicon from 'components/gridicon';
+
+interface Props {
+	content: TranslateResult;
+}
+
+const IncompatibleProductMessage: FC< Props > = ( { content } ) => {
```

Comment:
I'd propose this accept the `IncompatibleProducts` prop directly, and hide if there's no reason. That would prevent the pattern repeated in this code of `incompatibleProduct && <IncompatibleProductMessage />`.

Furthermore, in the spirit of separating presentation and logic, I feel that this component should contain the strings (presentation) and the selector below should pass them along (logic).